great_expectations/execution_engine/util.py

```diff
@@ -19,6 +19,48 @@
     WithinGroup = None
 
 
+# noinspection PyPep8Naming
+def get_or_create_spark_session(
+    name: str = "default_great_expectations_spark_dataframe_execution_engine",
+):
+    # Due to the uniqueness of SparkContext per JVM, it is impossible to change SparkSession configuration dynamically.
+    # Attempts to circumvent this constraint cause "ValueError: Cannot run multiple SparkContexts at once" to be thrown.
+    # Hence, SparkSession with SparkConf acceptable for all tests must be established at "pytest" collection time.
+    # This is preferred to calling "return SparkSession.builder.getOrCreate()", which will result in the setting
+    # ("spark.app.name", "pyspark-shell") remaining in SparkConf statically for the entire duration of the "pytest" run.
+    try:
+        from pyspark import SparkContext
+        from pyspark.sql import SparkSession
+
+        from great_expectations.execution_engine import SparkDFExecutionEngine
+    except ImportError:
+        SparkContext = None
+        SparkSession = None
+        SparkDFExecutionEngine = None
+        # TODO: review logging more detail here
+        logger.debug(
+            "Unable to load pyspark; install optional spark dependency for support."
+        )
+
+    execution_engine: SparkDFExecutionEngine = SparkDFExecutionEngine(
+        spark_config={
+            "spark.app.name": name,
+            "spark.sql.catalogImplementation": "hive",
+            "spark.executor.memory": "450m",
```

Comment:
???

Reply:
@jcampbell This is a default when the user DOES NOT specify spark_config -- and hence we have to come up with a default set of `spark_config` parameters.